eras/conway/impl/src/Cardano/Ledger/Conway/Governance.hs

```diff
@@ -727,3 +774,53 @@ votingDRepThresholdInternal pp isElectedCommittee action =
         ParameterChange _ ppu -> VotingThreshold $ pparamsUpdateThreshold pp ppu
         TreasuryWithdrawals {} -> VotingThreshold dvtTreasuryWithdrawal
         InfoAction {} -> NoVotingThreshold
+
+-- ===================================================================
+-- Lenses for access to
+-- 1. (DRepDistr (EraCrypto era))
+-- 2. The 3 inputs we need to initialize one
+--    a. (VMap VB VP (Credential 'Staking (EraCrypto era)) (CompactForm Coin)). Extracted from Incremental
+--    b. (Set (Credential 'DRepRole (EraCrypto era))). Registered DReps in the CertState
+--    c. (UMap (EraCrypto era)). The unified map in the DState.
+--                               We will use the to DRepUView to obtain  (Map (Credential 'Staking c) (DRep c))
+--                               to see what stake credentials delegate to which DReps
+-- 3. and its completion:  (Map (DRep c) (CompactForm Coin)).  The aggregated voting power of each DRep
+
+newEpochStateDRepDistrL :: ConwayEraGov era => Lens' (NewEpochState era) (DRepDistr (EraCrypto era))
+newEpochStateDRepDistrL = nesEsL . esLStateL . lsUTxOStateL . utxosGovStateL . drepDistrGovStateL
+
+epochStateDRepDistrL :: ConwayEraGov era => Lens' (EpochState era) (DRepDistr (EraCrypto era))
+epochStateDRepDistrL = esLStateL . lsUTxOStateL . utxosGovStateL . drepDistrGovStateL
+
+epochStateStakeDistrL ::
+  Lens'
+    (EpochState era)
+    (VMap VB VP (Credential 'Staking (EraCrypto era)) (CompactForm Coin))
+epochStateStakeDistrL = esSnapshotsL . ssStakeMarkL . ssStakeDistrL
+
+epochStateIncrStakeDistrL ::
+  Lens'
+    (EpochState era)
+    (Map (Credential 'Staking (EraCrypto era)) Coin)
+epochStateIncrStakeDistrL = esLStateL . lsUTxOStateL . utxosStakeDistrL . credMapL
+
+epochStateRegDrepL ::
+  Lens'
+    (EpochState era)
+    (Map (Credential 'DRepRole (EraCrypto era)) (DRepState (EraCrypto era)))
+epochStateRegDrepL = esLStateL . lsCertStateL . certVStateL . vsDRepsL
+
+epochStateUMapL :: Lens' (EpochState era) (UMap (EraCrypto era))
+epochStateUMapL = esLStateL . lsCertStateL . certDStateL . dsUnifiedL
```

Comment:
This lenses should not have been moved from: `Cardano.Ledger.Shelley.LedgerState.Types`, since they work directly with `EpochState`:

* epochStateStakeDistrL
* epochStateIncrStakeDistrL
* epochStateRegDrepL
* epochStateUMapL

Reply:
ok